### PR TITLE
Feat/profile mobile user interface

### PR DESCRIPTION
**Pull request by Wiecek-K**

### CODE REVIEW

**Review by NerdbordMentor — CHANGES_REQUESTED** [review]
Based on the provided task requirements and code changes, here is the code review:

1. **app/(profile)/my-profile/page.module.scss**: The changes in this file seem to be in line with the task requirements. The padding for mobile devices has been set to 0 which should help with the display on smaller screens.

2. **next.config.js**: The addition of 'sm.ign.com' to the list of image domains doesn't seem to be directly related to the task requirements. However, it doesn't contradict them either.

3. **src/backend/user/user.serializer.ts**: Commenting out the image property in the serialized user data could potentially cause issues if other parts of the application rely on this data. This change doesn't seem to be in line with the task requirements.

4. **src/components/CopyEmail/CopyEmail.module.scss**: The changes in this file should help improve the display of the CopyEmail component on mobile devices, which is in line with the task requirements.

5. **src/components/Headers/MyProfileHeader/MyProfileHeader.module.scss**: The changes in this file should help improve the display of the MyProfileHeader component on mobile devices, which is in line with the task requirements.

6. **src/components/MyProfile/ProfileDetails/ProfileDetails.module.scss**: The changes in this file should help improve the display of the ProfileDetails component on mobile devices, which is in line with the task requirements.

7. **src/components/MyProfile/ProfileDetails/ProfileDetails.tsx**: The hardcoded values for tech stack and bio are not in line with the task requirements. The data should be fetched from a backend or at least be dynamic.

8. **src/components/MyProfile/ProfileMain/ProfileMain.module.scss**: The changes in this file should help improve the display of the ProfileMain component on mobile devices, which is in line with the task requirements.

9. **src/components/MyProfile/ProfileMain/ProfileMain.tsx**: Similar to ProfileDetails.tsx, there are hardcoded values here that are not in line with the task requirements. The data should be fetched from a backend or at least be dynamic.

10. **src/components/MyProfile/ProfileMain/avatar-generations_prsz.jpg**: This file was added but it's not clear from the task requirements if a new avatar image was needed.

11. **src/inputs/LogOutBtn/LogOutBtn.module.scss**: The changes in this file should help improve the display of the LogOutBtn component on mobile devices, which is in line with the task requirements.

12. **src/utils/styleVariables.scss**: This file was added and it contains useful mixins for handling different screen sizes, which is in line with the task requirements.

In conclusion, most of the changes are in line with the task requirements of adjusting the Profile UI for mobile devices. However, there are some hardcoded values that need to be addressed.

File: src/utils/styleVariables.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  @media (max-width: 768px) {
+    @content;
+  }
+}
+
+@mixin desktop {
+  @media (min-width: 1280px) {
+    @content;
+  }
+}
```
Comment (wawrzonnn) [inline src/utils/styleVariables.scss:18]: We have a globals.scss file, that's where these mixins should be

File: src/components/MyProfile/ProfileMain/avatar-generations_prsz.jpg
Comment (wawrzonnn) [inline src/components/MyProfile/ProfileMain/avatar-generations_prsz.jpg:1]: This should be in the src/assets/images

File: src/backend/user/user.serializer.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ export const serializeUserToUserPayload = (data: UserWithRelations) => {
       id: data.githubDetails?.id,
       username: data.githubDetails?.username,
       userId: data.githubDetails?.userId,
-      image: data.githubDetails?.image,
+      // image: data.githubDetails?.image,
```
Comment (Sumick) [inline src/backend/user/user.serializer.ts:11]: **nitpick**: This can be removed

File: src/components/MyProfile/ProfileMain/ProfileMain.tsx [hunk 3]
```diff
@@ -26,32 +26,12 @@ const ProfileMain = async () => {
...
-              <CopyEmail email={session.user.email} />
-            </li>
-          </ul>
+          {/* //tutaj była ul z socialami */}
         </div>
         <div className={styles.profile}>
           <div className={styles.user}>
             <Image
-              src={session.user.image}
+              src="https://sm.ign.com/ign_nordic/cover/a/avatar-gen/avatar-generations_prsz.jpg"
```
Comment (Sumick) [inline src/components/MyProfile/ProfileMain/ProfileMain.tsx:34]: **nitpick:** Why?

**Review by Sumick — CHANGES_REQUESTED** [review]
I don't understand why did we reverted dynamic values to hardcoded ones here - if you needed them to set proper layout, I suggest you to mock profile, or create one

File: src/components/MyProfile/ProfileMain/ProfileMain.module.scss [hunk 4]
```diff
@@ -1,17 +1,36 @@
+@import '../../../utils/styleVariables.scss';
+
```
[the remark this answers is not in the document]
Comment (Wiecek-K) [reply src/components/MyProfile/ProfileMain/ProfileMain.module.scss:2]: image
I can't completely remove the imports.
I have to change them to global file path
